docs/api.rst

```diff
@@ -0,0 +1,149 @@
+API v2
+======
+
+The latest nodewatcher version uses API v2. Unlike v1 there is more control over which data is shown. In all of the examples below the wlan-si nodewatcher page is used for easier presentation. If you intend to use this on your own server replace the https://nodes.wlan-si.net with the address of your server. When you go to the API page in nodewatcher:
+
+https://nodes.wlan-si.net/api/v2/
+
+You are presented with certain groups of data:
+
+* ``pool/ip`` https://nodes.wlan-si.net/api/v2/pool/ip/
+* ``statistics/device`` https://nodes.wlan-si.net/api/v2/statistics/device/
+* ``link`` https://nodes.wlan-si.net/api/v2/link/
+* ``node`` https://nodes.wlan-si.net/api/v2/node/
+* ``project`` https://nodes.wlan-si.net/api/v2/project/
+* ``statistics/project`` https://nodes.wlan-si.net/api/v2/statistics/project/
+* ``statistics/status`` https://nodes.wlan-si.net/api/v2/statistics/status/
+* ``unknown_node`` https://nodes.wlan-si.net/api/v2/unknown_node/
+* ``user_authentication_key`` https://nodes.wlan-si.net/api/v2/user_authentication_key/
+* ``event`` https://nodes.wlan-si.net/api/v2/event/
+* ``warning`` https://nodes.wlan-si.net/api/v2/warning/
+* ``build_result`` https://nodes.wlan-si.net/api/v2/build_result/
+
+Querying data
+-------------
+
+We are going to focus on the ``node`` aspect. Some of the main features of requesting data through the API are:
+
+* ``format``
+* ``limit``
+* ``offset``
+* ``fields``
+* ``filters``
+
+Each option that you want to add to the API request is given in the query string (after the '?').
+
+Format
+------
+
+``format`` allows you to represent the gathered data in JSON format:
+
+https://nodes.wlan-si.net/api/v2/node/?format=JSON
+
+Limit
+-----
+
+``limit`` helps you to set an exact number of nodes that you want to show per page.
+
+https://nodes.wlan-si.net/api/v2/node/?limit=1
+
+This will give you a list of all registered nodes with each node on a single page.
+
+Offset
+------
+
+``offset`` allows you to go to a specific page of the API request. This alongside limit can allow you to show data for a certain node or set of nodes.
+
+https://nodes.wlan-si.net/api/v2/node/?offset=17
+
+Fields
+------
+
+When you make a simple API request like this https://nodes.wlan-si.net/API/v2/node/ you can see that there isn't much information for each of the nodes. That is because node data is divided in ``fields`` and you need to specify each ``field`` that you want to access. There are 2 groups ``config`` and ``monitoring``:
+
+* **config:**

```

Comment:
It will be hard to keep these lists up-to-date. Is there any way one could get this list programmatically? Through API? Or see it in the code we could link here?